cfgov/v1/jinja2/v1/layouts/layout-side-nav.html

```diff
@@ -23,3 +17,18 @@
     content__flush-top-on-small
     {{ 'content__half-top-on-desk' if breadcrumb_items | length > 0 else '' }}
 {%- endblock %}
+
+{% block javascript scoped %}
+    {{ super() }}
+    <script>
+      if ( document.body.parentElement.className.indexOf( 'no-js' ) === -1 ) {
+        !function() {
+          var s = [
+            '{{ static('js/routes/on-demand/secondary-navigation.js') }}'
+          ];
+          jsl(s);
+        }()
+      }
+    </script>
+{% endblock javascript %}
```

Comment:
@anselmbradford is there a more concise way to add a single file to the page JS, in a child template? This feels very verbose but seems to be the pattern we use [very often](https://github.com/cfpb/consumerfinance.gov/search?l=HTML&q=no-js&type=code) in the repo.

Is the `no-js` check needed? Wouldn't this code not even be executing if JS was disabled?

Reply:
`no-js` isn't related to a user disabling the JS in their browser, it's for browsers that support JS but are old and broken, so we deliver a "no JS" experience to them, which disables much of the JavaScript on the site except scripts coming in from analytics. By default we have `no-js` set on the `html` element ([here](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/v1/jinja2/v1/layouts/base.html#L24)), and then remove it if the visiting browser supports the [fetch API](https://caniuse.com/fetch) ([here](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/v1/jinja2/v1/layouts/base.html#L211)).

In the snippet here and elsewhere it's checking if the `html` element has the `no-js` class, and if it doesn't it dynamically loads in the JS script via the `jsl` ("JavaScript Loader"). **HOWEVER**, looking at the `jsl` definition, it will **only** load the script if the fetch API is supported, using the same logic that changes `no-js` to `js` ([here](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/v1/jinja2/v1/layouts/base.html#L202)). **So it does look like the check could be removed from here and in similar code blocks.**

Reply:
> So it does look like the check could be removed from here and in similar code blocks.

Thank you for the detailed explanation! How about if I open a separate PR removing those throughout?

Reply:
Opened #7531.